specification/maintenance/Maintenance.Management/ApplyUpdate.tsp

```diff
@@ -0,0 +1,144 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.Maintenance;
+/**
+ * Apply Update request
+ */
+@subscriptionResource
+model ApplyUpdate
+  is Azure.ResourceManager.ProxyResource<ApplyUpdateProperties> {
+  ...ResourceNameParameter<
+    Resource = ApplyUpdate,
+    KeyName = "applyUpdateName",
+    SegmentName = "applyUpdates",
+    NamePattern = ""
+  >;
+}
+
+@armResourceOperations
+interface ApplyUpdateOps
+  extends Azure.ResourceManager.Legacy.CustomLegacyOperations<
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+
+        /**
+         * Resource provider name
+         */
+        @path @segment("providers") providerName: string,
+
+        /**
+         * Resource parent type
+         */
+        @path resourceParentType: string,
```

Comment:
This is a really problematic pattern - it allows the extension to apply to any resource that is the direct child of a resource group resource.  It disallows any more deeply nexted resource, or a resource homed in the subscription or location.

IMO, we should consider correcting this API to use Scope - at the very least, we should add a set of scope-based operations